plugin/evm/syncervm_test.go

```diff
@@ -292,6 +293,12 @@ func createSyncServerAndClientVMs(t *testing.T, test syncTest) *syncVMSetup {
 	// configure [serverVM]
 	_, serverVM, _, serverAppSender := GenesisVM(t, true, genesisJSONSubnetEVM, "", "")
 	generateAndAcceptBlocks(t, serverVM, parentsToGet, func(i int, gen *core.BlockGen) {
+		b, err := results.NewPredicateResults().Bytes()
```

Comment:
if we move predicate result handling into the engine, this can be taken care of by `FinalizeAndAssemble` instead of requiring the extra modification here.

Reply:
I think that would be a nice change if possible.

Reply:
Looked into this and I think this would only be a worthwhile change if we also moved predicate verification into `Finalize` or at least into the engine for symmetry with `FinalizeAndAssemble`.

We could add engine callbacks in like with Coreth, but I generally don't think that was a great design, so planning to leave this as is in this PR.